src/routes/admin/_admin.tsx

```diff
@@ -129,6 +149,10 @@ function LayoutComponent() {
           <Sheet>
             <h1>SPV Wallet Admin</h1>
           </Sheet>
+          <Button variant="ghost" className="ml-auto" onClick={onRefreshClick}>
+            <RefreshCcw className="h-4 w-4 mr-2" ref={refreshRef} />
+            Refresh
+          </Button>
```

Comment:
I would do it more 'react-way'.
```js
const [isRefreshing, setIsRefreshing] = useState(false)
//...

const onRefreshClick = () => {
    setIsRefreshing(true)
    queryClient.invalidateQueries().finally(() => {
      setIsRefreshing(false)
    });
  };

//...
return (
//...
  <Button variant="ghost" className="ml-auto" onClick={onRefreshClick}>
    <RefreshCcw className={cx({
      "h-4 w-4 mr-2": true,
      'animate-spin': isRefreshing,
    })}/>
    Refresh
  </Button>
//...
)

```

I'm aware that we might not have the `cx` function - which is, I suppose, from `emotion`. But it's easy to implement exact or similar one.